Mitigate XS CI build flakes

I've found that occasionally the XS build crashes the GCC compiler, and returns an exit status 2.  I can replicate it by running:

```sh
status=0; while test $status -eq 0; do
  rm -rf $MODDABLE/build/tmp/lin/release/eventual-send
  mcconfig -m -p x-cli-lin test-xs-manifest.json
  status=$?
done; echo "Final status=$status"
```

This loop runs normally for the most part, but eventually crashes the mcconfig compile.

<details>
<summary>expected identifier at end of input</summary>
```
/home/runner/moddable/build/tmp/lin/release/marshal/mc.xs.c:2748:12: error: expected identifier at end of input
 /* 1460 */ { (txSlot*)&gxHeap[1461], {.ID = XS_NO_ID, .flag = 0xf0, .kind = XS_INSTANCE_KIND}, .value = { .instance = { NULL, (txSlot*)&gxHeap[656] } } },
            ^
[...]
##[error]Process completed with exit code 2.
```
</details>
<details>
<summary>excess elements in union initializer</summary>
```
/home/runner/moddable/build/tmp/lin/release/marshal/mc.xs.c:5059:0: warning: excess elements in union initializer
 /* 3771 */ { (txSlot*)&gxHeap[3772], {.ID = xsID_isNaN, .flag = 0x8e, .kind = XS_REFERENCE_KIND}, .value = { .reference = (txSlot*)&gxHeap[666] } },
[...]
##[error]Process completed with exit code 2.
```
</details>

<details>
<summary>error: ‘gxH’ undeclared here (not in a function)</summary>
```
/vagrant/moddable/build/tmp/lin/release/eventual-send/mc.xs.c:3711:121: error: ‘gxH’ undeclared here (not in a function)
 /* 2370 */ { NULL, {.ID = xsID_constructor, .flag = 0x8e, .kind = XS_REFERENCE_KIND}, .value = { .reference = (txSlot*)&gxHeap[2364] } },
                                                                                                                         ^~~
/vagrant/moddable/build/tmp/lin/release/eventual-send/mc.xs.c:3711:12: error: expected ‘}’ at end of input
 /* 2370 */ { NULL, {.ID = xsID_constructor, .flag = 0x8e, .kind = XS_REFERENCE_KIND}, .value = { .reference = (txSlot*)&gxHeap[2364] } },
            ^
/vagrant/moddable/build/tmp/lin/release/eventual-send/makefile:227: recipe for target '/vagrant/moddable/build/tmp/lin/release/eventual-send/mc.xs.c.o' failed
make: *** [/vagrant/moddable/build/tmp/lin/release/eventual-send/mc.xs.c.o] Error 1
make: *** Waiting for unfinished jobs....
Final status=2
```
</details>

With some shell-scriptery, the CI build can be a little more tolerant of these errors, and rerun the mcconfig steps if they fail.  That would help reduce the frequency of test flakes.

--- a/.github/workflows/ag-solo-xs.yml
+++ b/.github/workflows/ag-solo-xs.yml
@@ -37,6 +37,27 @@ jobs:
         mv tape-xs-ag03 tape-xs
         cd tape-xs
         yarn install
+    - name: create /usr/local/bin/noflake
+      run: |
+        nf=/usr/local/bin/noflake
+        sudo tee "$nf" <<\EOF >/dev/null
+        #! /bin/sh
+        # noflake: rerun a command until it doesn't flake out
+        CMD=${1+"$@"}
+        FLAKE_STATUS=2
+        RETRIES=3
+        status=$FLAKE_STATUS
+        tries=0
+        while test $tries -le $RETRIES -a $status -eq $FLAKE_STATUS; do
+          tries=`expr $tries + 1`
+          echo 1>&2 "noflake: try number $tries of: $CMD"
+          $CMD
+          status=$?
+          echo 1>&2 "noflake: exit status $status"
+        done
+        exit $status
+        EOF
+        sudo chmod +x "$nf"
     - name: test eventual-send on xs
       run: |
         export MODDABLE=$HOME/moddable
@@ -44,7 +65,7 @@ jobs:
         export TAPE=$HOME/tape-xs
         cd packages/eventual-send
         node -r esm $TAPE/bin/tape-xs-build.js $PWD test/test*.js
-        mcconfig -m -p x-cli-lin test-xs-manifest.json
+        noflake mcconfig -m -p x-cli-lin test-xs-manifest.json
         $MODDABLE/build/bin/lin/release/eventual-send
     - name: test marshal on xs
       run: |
@@ -53,5 +74,5 @@ jobs:
         export TAPE=$HOME/tape-xs
         cd packages/marshal
         node -r esm $TAPE/bin/tape-xs-build.js $PWD test/test*.js
-        mcconfig -m -p x-cli-lin test-xs-manifest.json
+        noflake mcconfig -m -p x-cli-lin test-xs-manifest.json
         $MODDABLE/build/bin/lin/release/marshal

